### PR TITLE
Kernel API `/api/file/readDir` support for returning symbolic link information

### PR DESCRIPTION
**Pull request by Zuoqiu-Yingyi**
* [x] Please commit to the dev branch
* [x] For contributing new features, please supplement and improve the corresponding user guide documents

- directory symlink: `isDir: true, isSymlink: true`
- directory normal: `isDir: true, isSymlink: false`
- file symlink: `isDir: false, isSymlink: true`
- file normal: `isDir: false, isSymlink: false`

### CODE REVIEW

File: kernel/api/file.go [hunk 1]
```diff
@@ -172,9 +172,18 @@ func readDir(c *gin.Context) {
...
+			logging.LogErrorf("stat [%s] failed: %s", path, err)
+			ret.Code = 500
+			ret.Msg = err.Error()
+			return
+		}
 		files = append(files, map[string]interface{}{
-			"name":  entry.Name(),
-			"isDir": entry.IsDir(),
+			"name":      entry.Name(),
+			"isDir":     info.IsDir(),
```
Comment (fenilgmehta) [inline kernel/api/file.go:185]:
```suggestion
			"isDir":     entry.IsDir(),
```
Why change from `entry.IsDir()` to `info.IsDir()` ?
`entry.IsDir()` / `dir.IsDir()` (where `entry` and `dir` are of type `fs.DirEntry`) is being used in `kernel/util/file.go > IsDirRegularOrSymlink` too.

Comment (Zuoqiu-Yingyi) [reply kernel/api/file.go:185]:
```go
// entry is a symlink to a dir
path := filepath.Join(dirPath, entry.Name())
info, err = os.Stat(path)

entry.IsDir() // return false
info.IsDir() // return true
```

Comment (88250) [reply kernel/api/file.go:185]: Guys, let's make sure the code here is ok:

image

Comment (88250) [reply kernel/api/file.go:185]: Oh, sorry, I missed the changes in file.go later, I have extracted a function `IsSymlink()`, it seems to be no problem.

Comment (Zuoqiu-Yingyi) [reply kernel/api/file.go:185]: > Guys, let's make sure the code here is ok:
> 
> image

If this function's name is `IsDirRegularOrSymlink`, it's Ok.
If this function's name is `IsDirRegularOrDirSymlink`, it's not Ok.👀

**Review by 88250 — APPROVED** [review]

File: kernel/api/file.go (hunk 1 above)
Comment (88250) [reply kernel/api/file.go:185]: @Zuoqiu-Yingyi Thanks for the explanation, I get it, merge first.

@fenilgmehta If there is something we missed, please continue to add comments, thank you.

**Pull request merged**